### PR TITLE
Use ZoneOffset Rather Than ZoneId In Most Arbitraries

### PR DESCRIPTION
**Pull request by isomarcte**
This works around special case rules for ZoneId values which cause issues with testing. We still test ZoneId itself, but we use ZoneOffset to construct the various time values.

This commit also change the `Arbitrary` `ZoneOffset` to be based of the ScalaCheck implementation which fully covers the domain. We can remove this once https://github.com/circe/circe/issues/1994 is complete.

### CODE REVIEW

**Review by armanbilge — APPROVED** [review]

File: modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala [hunk 1]
```diff
@@ -25,6 +26,28 @@ class JavaTimeCodecSuite extends CirceMunitSuite {
...
+    new Choose[ZoneOffset] {
+      override def choose(min: ZoneOffset, max: ZoneOffset): Gen[ZoneOffset] =
+        min.compareTo(max) match {
+          case 0                    => Gen.const(min)
+          case result if result > 0 => Gen.fail
+          case _                    =>
+            // Looks flipped, but it is not.
+            Gen.choose(max.getTotalSeconds, min.getTotalSeconds).map(value => ZoneOffset.ofTotalSeconds(value))
+        }
+    }
```
Comment (zarthross) [inline modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala:41]: I'm fine with bumping scalacheck to 1.16.0 if we need to so we don't have to do this backport.

**Review by zarthross — CHANGES_REQUESTED** [review]

File: modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala (hunk 1 above)
Comment (armanbilge) [reply modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala:41]: We need ScalaCheck 1.17.0 actually.

**Comment by isomarcte** [comment]
@zarthross the failure in CI is fixed in #1992. I'll merge with main.

File: modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala (hunk 1 above)
Comment (zarthross) [reply modules/tests/shared/src/test/scala/io/circe/JavaTimeCodecSuite.scala:41]: Ok, we can keep this backported until 1.17.x then.

**Pull request merged**